packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -192,14 +192,20 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	): [change: TChange, newCommit: GraphCommit<TChange>] {
 		this.assertNotDisposed();
 
+		const changeWithRevision = this.changeFamily.rebaser.replaceRevisions(
```

Comment:
I'm wondering if it would be better to have the changeset carry the revision from the start (which would make this replacement unnecessary).
I think that might entail changing where the revision tag is minted.
Thoughts?

Reply:
I think we can do this in a follow up PR.

Reply:
Agreed.